api/v2/views/instance_action.py

```diff
@@ -0,0 +1,37 @@
+import django_filters
+import uuid
+
+from rest_framework import filters
+
+from api.v2.serializers.details import InstanceActionSerializer
+from api.v2.views.base import AuthReadOnlyViewSet
+from core.models import InstanceAction, Instance
+
+
+class InstanceActionFilter(filters.FilterSet):
+    provider_id = django_filters.MethodFilter(action='filter_by_provider')
+    instance_id = django_filters.MethodFilter(action='filter_by_instance')
+
+    def filter_by_instance(self, queryset, value):
+        """
+        Filter actions down to those available for a specific instance
+        """
+        return InstanceAction.filter_by_instance(value, queryset)
+
+    def filter_by_provider(self, queryset, value):
+        """
+        Filter actions down to those available for a specific provider
+        """
+        return InstanceAction.filter_by_provider(value, queryset)
```

Comment:
Yes, `filter_by_provider` would be where you enable/disable functionality, for example, disable shelving on the Tucson cloud.